theano/gpuarray/c_code/dnn_gw.c

```diff
@@ -305,6 +348,17 @@ APPLY_SPECIFIC(conv_gw)(PyGpuArrayObject *input, PyGpuArrayObject *output,
             hashkey.c_str()
      );
 #endif
+#ifdef DEBUG_TIMING
+    if (!(reuse_algo || use_cached)) {
```

Comment:
Here, if we always reuse one shape, we will never print the timing of computation after the first call. What about printing it for every n_section%100? or  1000 or something like that?

Ideally, it would be to print  it at the last call, but I don't know how to do that.

Do you think this is overkill and it won't be used?

Reply:
This code block prints timing info about algorithm selection, thus if we always reuse one shape, it will be printed at the first call (when selection is done) but not on other calls (as there is no selection in next calls), to avoid overload the console output.

Timing info about computation itself will be always printed (see lines below just before the final `return 0;`).

Is it acceptable ?

Reply:
I would go with that for now. Maybe we could n_selection and/or the selection time to the computation line. We will see.